website/content/docs/upgrading/instructions/testing-experimental-raft-backend.mdx

```diff
@@ -0,0 +1,420 @@
+  leave it running for a few days or weeks to gain confidence before attempting
+  it in production.
+
+## Risks
+
+Although we are describing a way to test this that limits risk as far as
+possible, there still are potential risks it's important to know:
+
+ - If WAL is enabled on a server and is found to corrupt data in some way. That
+   server's data can't be recovered. The server will need to be restarted with
+   an empty data directory and reload it's state from the leader.
+ - It's possible that WAL might corrupt data or otherwise have a bug that causes
+   the server to panic and crash. It may even not be able to restart if the same
+   bug occurs when it reads from the logs on startup. In this case as above the
+   server would need to be restarted with WAL disabled and an empty data
+   directory.
+ - It's _possible_ though unlikely that if WAL corrupted data, clients might
+   read corrupted data from that server. For example invalid IP addresses or
+   have tokens fail to match. This is unlikely even if there is a WAL corruption
+   bug hit because replication typically takes place using in-memory cached
+   objects rather than reads from disk. The issue would be fixed by restoring
+   the server.
+ - If you enable a server to use WAL using Consul OSS or on a voting server with
+   Consul Enterprise, it's _possible_ that the WAL could cause corruption of
+   that server's state (with the caveats above) _and then_ become the leader and
+   replicate that corruption to all other servers. In this scenario only a
+   restore from backup would recover a completely un-corrupt state.
+   - If you test on a non-voting server in Enterprise, this can't happen.
+
+## Procedure to enable WAL on one server
+
+**1. Enable log verification (new in 1.15).**
+
+This needs to be enabled on any voting server in Enterprise and all servers in
+OSS since it is the leader that writes verification checkpoints.
+
+On each voting server add the following to the server's configuration file:
+
+```hcl
+raft_logstore {
+  verification {
+    enabled = true
+    interval = "60s"
+  }
+}
+```
+
+You'll need to restart each server in turn for that to take effect (`consul
+reload` is not sufficient). Wait for each one to become a healthy voter again
+using `consul operator raft list-peers` before moving on to the next. This can
+take a few minutes if the snapshot is large.
+
+You should now see log entries on the servers every minute that look like this:
+
+```
+2023-01-31T14:44:31.174Z [INFO]  agent.server.raft.logstore.verifier: verification checksum OK: elapsed=488.463268ms leaderChecksum=f15db83976f2328c rangeEnd=357802 rangeStart=298132 readChecksum=f15db83976f2328c
+```
+
+
+**2. Select a server to enable WAL on.**
+
+As noted in [Risks](#risks), Consul Enterprise users should select a non-voting
+server at first. For Consul OSS users, or Enterprise users who don't have
+non-voting servers, select one of the follower servers.
```

Comment:
Shall we hint on the fact that ENT users could test it by adding a new non-voting server in case they don't have one deployed yet? This will reduce their risks.